### PR TITLE
[YUNIKORN-2448] Implemented licenses modal

### PR DESCRIPTION
**Pull request by dcoric**
### What is this PR for?
This PR adds a menu option to display licenses used by the project. Clicking that menu option opens a modal dialog where all licenses are listed.

A list of licenses is generated on build time. For dev preview, a short example file is used.

### What type of PR is it?
* [ ]  - Bug Fix
* [x]  - Improvement
* [ ]  - Feature
* [ ]  - Documentation
* [ ]  - Hot Fix
* [ ]  - Refactoring

### What is the Jira issue?
[YUNIKORN-2448](https://issues.apache.org/jira/browse/YUNIKORN-2448)

### Questions?
* Should we append at the beginning of the licenses preview dialog information that the project is licensed under the Apache License 2.0?

### Screenshots (if appropriate)
Screenshot 2024-03-04 at 22 39 34
Screenshot 2024-03-04 at 22 39 52

### CODE REVIEW

**Review by chia7712 — APPROVED** [review]
LGTM

File: src/app/components/licenses-modal/licenses-modal.component.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+export class LicensesModalComponent {
+  licenses = '';
+
+  constructor(
+    private dialogRef: MatDialogRef<LicensesModalComponent>,
+    private httpClient: HttpClient
+  ) {
+    // Licenses file is generated during build time and is available at /3rdpartylicenses.txt
+    // Dev mode uses the file from assets folder assets/example-licenses.txt
+    const licensePath = isDevMode() ? 'assets/example-licenses.txt' : '3rdpartylicenses.txt';
```
Comment (chia7712) [inline src/app/components/licenses-modal/licenses-modal.component.ts:38]: Could we use `LICENSE` file to replace `assets/example-licenses.txt`?

Comment (dcoric) [reply src/app/components/licenses-modal/licenses-modal.component.ts]: That is a better solution. I have updated the PR with this change.
Thanks!

**Comment by chia7712** [comment]
@dcoric thanks for this contribution!

**Pull request closed without merging**